docs/src/cookbooks/cookbook.md

```diff
@@ -157,6 +158,39 @@ See the [DataView cookbook](dataview#how-do-i-connect-a-subset-of-bundle-fields)
 
 ## Vectors and Registers
 
+### Can I make a 2D or 3D Vector?
+
+Yes. Using `VecInit` you can make Vectors that hold Vectors of Chisel types. Methods `fill` and `tabulate` make these multi-dimensional Vectors.
+
+```scala mdoc:silent:reset
+import chisel3._
+
+class MyBundle extends Bundle {
+  val foo = UInt(4.W)
+  val bar = UInt(4.W)
+}
+
+class Foo extends Module {
+  
+  val twoDVec = VecInit.fill(2,3)(5.U)
+  val threeDVec = VecInit.fill(1,2,4)(new MyBundle)
+
+  val indexTiedVec = VecInit.tabulate(2,2){ (x, y) => (x+y).asUInt }
+  assert(indexTiedVec(0)(0) === 0.U)
+  assert(indexTiedVec(0)(1) === 1.U)
+  assert(indexTiedVec(1)(0) === 1.U)
+  assert(indexTiedVec(1)(1) === 2.U)
+
+  val indexTiedVec3D = VecInit.tabulate(1,2,3){ (x, y, z) => (x*y*z).asUInt }
+  assert(indexTiedVec(0)(0)(0) === 0.U)
+  assert(indexTiedVec(1)(1)(1) === 1.U)
+  assert(indexTiedVec(1)(1)(2) === 2.U)
+  assert(indexTiedVec(1)(1)(3) === 3.U)
+  assert(indexTiedVec(1)(2)(3) === 6.U)
+
+}
+```
+
```

Comment:
It would be good to actually run this test too:

````suggestion
```scala mdoc:invisible
// Hidden but will make sure this actually compiles
ChiselStage.emitVerilog(new Foo)
```
````